rasa/core/domain.py

```diff
@@ -133,8 +152,9 @@ def from_yaml(cls, yaml: Text) -> "Domain":
         return cls.from_dict(data)
 
     @classmethod
-    def from_dict(cls, data: Dict) -> "Domain":
-        utter_templates = cls.collect_templates(data.get("responses", {}))
+    def from_dict(cls, data: Dict) -> Optional["Domain"]:
```

Comment:
Why is it now `Optional`? Looks like we're always returning `cls(...)`

Reply:
that's a good point, I think I messed up.